### ISSUE TITLE
Use webhooks to decide when to enforce in repos

### ISSUE THREAD

**Issue by ghost** [ossf/allstar#242]
Creating this as a tracking issue. Jeff has a design for the feature as mentioned in #89. Implementing this feature is low priority currently.

**Comment by ghost** [comment]
I wanted to add a note that the Scorecard Binary Artifacts check has an exemption for `gradle-wrapper.jar` files which are validated by [gradle/wrapper-validation-action](https://github.com/gradle/wrapper-validation-action) as part of a passing Workflow, but if Allstar enforcement happens immediately upon push, those checks may consistently be incomplete (instead of "success" status), eliminating the exemption.

One solution might be to leave Binary Artifacts as only a periodic check. Another might be to have it run after all Workflow runs on the latest commit are complete.

**Comment by emtunc** [comment]
Is this issue about having allstar trigger on webhook events or something else? I was going to create a FR for allstar to act when a repo/org webhook has been triggered as opposed to scanning every X hours but then I saw this and wasn't sure if it's the same thing

**Comment by jeffmendoza** [comment]
Yes, This is it. We want to just do the scan when we receive a webhook from github that something has changed. I've put some thought into this, and have some design ideas that I will post, Likely next week.
